### PR TITLE
[libc] Remove specific nan payload in math functions

### PR DESCRIPTION
**Pull request by gchatelet**

### CODE REVIEW

File: libc/src/math/generic/asinf.cpp [hunk 1]
```diff
@@ -109,7 +109,7 @@ LLVM_LIBC_FUNCTION(float, asinf, (float x)) {
       fputil::set_errno_if_required(EDOM);
       fputil::raise_except_if_required(FE_INVALID);
     }
-    return x + FPBits::build_nan(Sign::POS, FPBits::FRACTION_MASK).get_val();
+    return x + FPBits::build_nan().get_val();
```
Comment (gchatelet) [inline libc/src/math/generic/asinf.cpp:112]: @lntue independent from this patch, I don't really understand why we need to add `x` and `NaN`. Is there a reason?

File: libc/src/math/generic/sincosf.cpp [hunk 2]
```diff
@@ -148,9 +148,7 @@ LLVM_LIBC_FUNCTION(void, sincosf, (float x, float *sinp, float *cosp)) {
       fputil::set_errno_if_required(EDOM);
       fputil::raise_except_if_required(FE_INVALID);
     }
-    *sinp =
-        x +
-        FPBits::build_nan(fputil::Sign::POS, FPBits::FRACTION_MASK).get_val();
+    *sinp = x + FPBits::build_nan().get_val();
```
Comment (gchatelet) [inline libc/src/math/generic/sincosf.cpp:151]: same here

File: libc/src/math/generic/asinf.cpp (hunk 1 above)
Comment (lntue) [reply libc/src/math/generic/asinf.cpp:112]: It was carried from the old implementations, where we didn't have `build_quiet_nan` yet.  They should all be replaced by `build_quiet_nan`.

**Comment by gchatelet** [comment]
@lntue can you PTAL?
If this patch is correct this means we can completely remove `FPBits::build_nan`

**Review by lntue — APPROVED** [review]

**Pull request merged**